api/envoy/extensions/transport_sockets/tls/v3/tls.proto

```diff
@@ -109,8 +130,14 @@ message CommonTlsContext {
         [(validate.rules).message = {required: true}];
 
     // Config for fetching validation context via SDS API.
-    SdsSecretConfig validation_context_sds_secret_config = 2
-        [(validate.rules).message = {required: true}];
+    SdsSecretConfig validation_context_sds_secret_config = 2 [
+      (validate.rules).message = {required: true},
+      (udpa.annotations.field_migrate).oneof_promotion = "dynamic_validation_context"
+    ];
+
+    // Config for fetching validation context via CA Manager plugin
+    CAClientPluginConfig validation_context_ca_manager_plugin = 3
```

Comment:
This should also have `[#not-implemented-hide:]` as it's not part of Envoy yet. Arguably, there should be an Envoy-side implementation added.

Reply:
I will add the annotation. Just curious: so "not-implemented-hide" means not implemented in Envoy even if other clients (e.g. gRPC) implement it?

I can't do much about an Envoy-side implementation.

Reply:
https://github.com/envoyproxy/envoy/issues/11085 will make this better in the future.